src/settings.cpp

```diff
@@ -527,18 +527,27 @@ static void Write_ValidateString(void *ptr, const SaveLoad *sld, const char *p)
 /**
  * Set the string value of a setting.
  * @param ptr Pointer to the std::string.
- * @param sld Pointer to the information for the conversions and limitations to apply.
+ * @param sd  Pointer to the information for the conversions and limitations to apply.
  * @param p   The string to save.
  */
-static void Write_ValidateStdString(void *ptr, const SaveLoad *sld, const char *p)
+static void Write_ValidateStdString(void *ptr, const SettingDesc *sd, const char *p)
 {
 	std::string *dst = reinterpret_cast<std::string *>(ptr);
 
-	switch (GetVarMemType(sld->conv)) {
+	switch (GetVarMemType(sd->save.conv)) {
 		case SLE_VAR_STR:
 		case SLE_VAR_STRQ:
 			if (p != nullptr) {
-				dst->assign(p);
+				if (sd->desc.max != 0 && dst->size() >= sd->desc.max) {
```

Comment:
I still don't understand this line. What is "dst->size()"?
"dst" is only assigned 4 lines down, so size() ist always zero here?

Reply:
It was broken, it had to be `strlen(p)`.